third_party/txt/src/txt/paragraph.cc

```diff
@@ -552,6 +553,15 @@ void Paragraph::Layout(double width, bool force) {
                                bidi_run.direction(), bidi_run.style());
       }
     }
+    bool line_runs_all_rtl =
```

Comment:
What happens if only some of the runs are RTL?

Reply:
You'll get the current behavior, which may not calculate the justification correctly.  This will require further investigation.

Reply:
This won't regress it at least. Sounds good.